draft-ietf-add-split-horizon-authority.xml

```diff
@@ -382,28 +382,36 @@
 
     <section>
       <name>Delegating DNSSEC across Split DNS Boundaries</name>
-      <t>We wish to enable DNSSEC validation of local DNS names without requiring
-      the local resolver to hold DNSSEC private keys that are valid for the parent
+      <t>When the local zone can be signed with globally trusted keys for the parent
+      zone, support for DNSSEC can be accomplished simply by placing a zone cut at
+      the parent zone and including a suitable DS record for the local resolver's
+      DNSKEY.  Zones in this configuration appear the same to validating stubs whether
+      or not they implement this specification.</t>
+      <t>We also wish to enable DNSSEC validation of local DNS names when the local
+      zone cannot safely be signed with DNSSEC private keys that are valid for the parent
       zone.  To support this configuration, parent zones <bcp14>MAY</bcp14> add a
       "ds=..." key to the Verification Record whose value is the RDATA of a single
       DS record, base64url-encoded. This DS record authorizes a DNSKEY whose Owner
-      Name is "resolver.arpa."</t>
+      Name is "resolver.arpa.".</t>
       <t>To validate DNSSEC, the client first fetches and validates the Verification
       Record.  If it is valid and contains a "ds" key, the client <bcp14>MAY</bcp14>
       send a DNSKEY query for "resolver.arpa." to the local encrypted resolver.
       At least one resulting DNSKEY RR <bcp14>MUST</bcp14> match the DS RDATA from
       the "ds" key in the Verification Record. All local resolution results for
-      subdomains in this claim <bcp14>MUST</bcp14> offer RRSIGs that chain to one
-      of these approved DNSKEYs.</t>
+      subdomains in this claim <bcp14>MUST</bcp14> offer RRSIGs that chain to a
+      DNSKEY whose RDATA is identical to one of these approved DNSKEYs.</t>
       <t>The "ds" key <bcp14>MAY</bcp14> appear multiple
       times in a single Verification Record, in order to authorize multiple DNSKEYs
       for this local encrypted resolver.  If the "ds" key is not present in a valid
       Verification Record, the client <bcp14>MUST</bcp14> disable DNSSEC validation
       when resolving the claimed subdomains via this local encrypted resolver.</t>
+      <t>Note that in this configuration, any claimed subdomains MUST be marked as
+      unsigned in the public DNS.  Otherwise, resolution results would be rejected
+      by validating stubs that do not implement this specification.</t>
       <figure>
         <name>Example use of "ds=..."</name>
         <sourcecode>
-;; Parent zone
+;; Parent zone.
```

Comment:
This example is showing the case where compatibility with non-split-aware validating stubs is achieved by marking the delegation as unsigned, so there is no DS record for subdomain.parent.example.

If an actual DS RRset exists, then the "ds=..." key has no utility.  For the sake of non-split-aware validating stubs, all records must be valid under the DS RRset, and the "ds=..." key is not needed.

Reply:
Hi Ben,

Yes, I see the example works for non-split-aware validating stubs that do not use network-designated resolver. what would happen in case of a DNSSEC validating stub that uses network-designated resolver but has not implemented this specification. In the latter case, DNSSEC validation would fail for stubs that use network-designated resolver. Do we need to mention it ?



Cheers,
-Tiru

Reply:
DNSSEC validation doesn't have a concept of "fail".  There are [four possible resolution results](https://datatracker.ietf.org/doc/html/rfc4033#section-5): Secure, Insecure, Bogus, and Indeterminate.  If the records are not signed with a standard globally valid signature chain, then the best we can do for legacy stub resolvers is "Insecure", which is what is shown in this example.  The records will be seen as insecure because there is a globally valid, signed NSEC record proving that a DS record for the claimed zone does not exist.

Reply:
Got it, Thanks.